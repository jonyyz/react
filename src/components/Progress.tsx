import * as React from "react";

import {
  createStyles,
  withStyles
} from "@material-ui/core/styles";
import CircularProgress from "@material-ui/core/CircularProgress";

const styles = createStyles({
  container: {
    display: "flex",
    flexDirection: "column",
    flexGrow: 1,
    alignItems: "center",
    justifyContent: "center",
  },
});

interface IProps {
  classes: Record<string, string>
}

const Progress: React.FunctionComponent<IProps> = (props) => {
  const { classes } = props;

  return (
    <div className={classes.container}>
      <CircularProgress size={50} thickness={6} />
    </div>
  );
}

export default withStyles(styles)(Progress);
