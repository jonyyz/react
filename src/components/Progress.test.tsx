import * as React from "react";
import { shallow } from "enzyme";

import Progress from "./Progress";

describe("Progress", () => {
  test("should render", () => {
    const wrapper = shallow(<Progress />);
    expect(wrapper).toMatchSnapshot();
  });
});
