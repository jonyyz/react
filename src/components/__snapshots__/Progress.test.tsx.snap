// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress should render 1`] = `
<div
  className="makeStyles-container-1"
>
  <WithStyles(ForwardRef(CircularProgress))
    size={50}
    thickness={6}
  />
</div>
`;
